deployment/hexfunctions.go

```diff
@@ -93,25 +82,34 @@ func Cube2XY(cube vlib.Location3D, hexsize float64) vlib.Location3D {
 	return result
 }
 
-func ClosestMirror(mirrorTable []vlib.Location3D, pt vlib.Location3D) (distance float64, indx int, dv vlib.Location3D) {
-	distance = 15000
-	indx = -1
-
-	for i, c := range mirrorTable {
-		src := Cube2XY(pt, 1)
-		dest := Cube2XY(c, 1)
-		dc := dest.DistanceFrom(src)
-		// dc := (math.Abs(pt.X-c.X) + math.Abs(pt.Y-c.Y) + math.Abs(pt.Z-c.Z)) / 2.0
-		// dc := pt.DistanceFrom(c)
-
-		// log.Println("Index %d", i, c, dc)
-		if distance > dc {
-			distance = dc
-			indx = i
-			dv.SetXYZ(pt.X-c.X, pt.Y-c.Y, pt.Z-c.Z)
+func gapmeasure(hexsize float64) float64 {
+	return hexsize * math.Sqrt(19)
+}
+
+func ClosestMirror(mirrorTable map[vlib.Location3D]int, pt vlib.Location3D, gap float64, prev int, ISD float64) int {
+	vid := -10
+	var center vlib.Location3D
+	for c, i := range mirrorTable {
+		src := Cube2XY(pt, ISD/math.Sqrt(3))
+		src = vlib.FromCmplx(src.Cmplx()*vlib.GetEJtheta(30) + center.Cmplx())
+		distance := c.Distance2DFrom(src)
+		// fmt.Println("source , destination:", src, c)
+		// fmt.Println("gap,distance:", gap, distance)
+		if math.Abs(gap-distance) < 1 {
+			if (src.X > 0 && src.Y > 0) || (src.X > 0 && math.Abs(src.Y) < 1) {
+				if (math.Abs(c.Y-src.Y)-ISD) < 1 && (math.Abs(c.X-src.X)-4*ISD) < 1 {
+					vid = i
+					return vid
+				}
+			}
+			if prev != i {
+				vid = i
+
+			}
+
 		}
 	}
-	return distance, indx, dv
+	return vid
 }
```

Comment:
non backward compatible changes !! function arguments are different

Reply:
Created new function for the same keeping the old as it was.